astropy/coordinates/representation.py

```diff
@@ -1762,6 +1829,19 @@ def norm(self):
     def _combine_operation(self, op, other, reverse=False):
         return NotImplemented
 
+    def transform(self, matrix):
+        """Radial representations cannot be transformed by a Cartesian matrix.
+
+        Raises
+        ------
+        NotImplementedError
+
+        """
+        raise NotImplementedError(
+            "Radial representations cannot be transformed "
+            "by a Cartesian matrix."
+        )
```

Comment:
@mhvk  Potential followup: only $a * I_3$ ( multiplication) is an allowed transformation. That would be more consistent because there is an equivalent `__mul__` method

Reply:
Indeed, that makes sense. But since really `RadialRepresentation` has little purpose beyond being the base class for the differential, fine to leave it for follow-up!

Reply:
#11576